src/auth/certificatesubjecttokensupplier.ts

```diff
@@ -0,0 +1,299 @@
+  /**
+   * If true, uses the default well-known location for the certificate config.
+   * Either this or `certificateConfigLocation` must be provided.
+   */
+  useDefaultCertificateConfig?: boolean;
+  /**
+   * The file path to the certificate configuration JSON file.
+   * Required if `useDefaultCertificateConfig` is not true.
+   */
+  certificateConfigLocation?: string;
+  /**
+   * The file path to the trust chain (PEM format). If not provided,
+   * only the leaf certificate will be sent.
+   */
+  trustChainPath?: string;
+}
+
+/**
+ * Represents the "workload" block within the certificate configuration file.
+ * @internal
+ */
+interface WorkloadCertConfigJson {
+  cert_path: string;
+  key_path: string;
+}
+
+/**
+ * Represents the structure of the certificate_config.json file.
+ * @internal
+ */
+interface CertificateConfigFileJson {
+  version: number;
+  cert_configs: {
+    workload?: WorkloadCertConfigJson;
+  };
+}
+
+// --- Main Class ---
+
+/**
+ * A subject token supplier that uses a client certificate for authentication.
+ * It provides the certificate chain as the subject token for identity federation.
+ */
+export class CertificateSubjectTokenSupplier implements SubjectTokenSupplier {
+  private readonly certificateConfigPath: string;
+  private readonly trustChainPath?: string;
+  private readonly certPath: string;
+  private readonly keyPath: string;
+
+  /**
+   * Initializes a new instance of the CertificateSubjectTokenSupplier.
+   * @param opts The configuration options for the supplier.
+   */
+  constructor(opts: CertificateSubjectTokenSupplierOptions) {
+    if (!opts.useDefaultCertificateConfig && !opts.certificateConfigLocation) {
+      throw new InvalidConfigurationError(
+        'Either `useDefaultCertificateConfig` must be true or a `certificateConfigLocation` must be provided.',
+      );
+    }
+    if (opts.useDefaultCertificateConfig && opts.certificateConfigLocation) {
+      throw new InvalidConfigurationError(
+        'Both `useDefaultCertificateConfig` and `certificateConfigLocation` cannot be provided.',
+      );
+    }
+    this.trustChainPath = opts.trustChainPath;
+    this.certificateConfigPath = this.#resolveCertificateConfigFilePath(
+      opts.certificateConfigLocation,
+    );
+    ({certPath: this.certPath, keyPath: this.keyPath} =
+      this.#getWorkloadCertificatePaths(this.certificateConfigPath));
+  }
+
+  /**
+   * Creates an HTTPS agent configured with the client certificate and private key for mTLS.
+   * @returns An mTLS-configured https.Agent.
+   */
+  public createMtlsHttpsAgent(): https.Agent {
+    try {
+      const cert = fs.readFileSync(this.certPath);
+      const key = fs.readFileSync(this.keyPath);
+      return new https.Agent({key, cert});
+    } catch (err) {
+      const message = err instanceof Error ? err.message : String(err);
+      throw new CertificateSourceUnavailableError(
+        `Failed to read certificate or key file: ${message}`,
+      );
```

Comment:
Suggestion: it might be helpful to include the file path in the error message. We could use two different try/catch blocks for reading the key and the certificate files, so we know which file caused the error.